File: pulse-go/internal/pulse/color_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestColor(t *testing.T) {
	t.Run("Colors array should be valid", func(t *testing.T) {
		for color := range Colors {
			if Colors[color] != color {
				t.Errorf("wanted colors at index %d to be %v, got %v", color, color, Colors[color])
			}
		}
	})
}

File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/ColorTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.colors
import kotlin.test.Test
import kotlin.test.assertEquals

class ColorTest {
	@Test
	fun `Colors array should be valid`() {
		for (color in colors.indices) {
			assertEquals(colors[color], color)
		}
	}
}
